src/pca_report_library/__init__.py

```diff
@@ -0,0 +1,16 @@
+"""The pca-report-library library."""
+# We disable a Flake8 check for "Module imported but unused (F401)" here because
+# although this import is not directly used, it populates the value
+# package_name.__version__, which is used to get version information about this
+# Python package.
+from ._version import __version__  # noqa: F401
+
+__all__ = [
+    "compiler.xelatex",
+    "customer.closing",
+    "customer.generate_report",
+    "customer.graphs",
+    "templates.generate_template",
+    "utility.gets",
+    "utility.time",
+]
```

Comment:
These are not valid values for `__all__`.

```console
>>> from pca_report_library import *
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: module 'pca_report_library' has no attribute 'compiler.xelatex'
```

Reply:
> Thank you for continued work in dragging this old project into a modern development state.
> 
> I would like all of the `__init__.py` files for sub-modules to get the following boilerplate added:
> 
> ```python
> # Standard Python Libraries
> from typing import List
> 
> __all__: List[str] = []
> ```
> 
> Please remove the
> 
> ```python
> if __name__ == "__main__":
>     sys.exit(main())
> ```
> 
> bits from the end of `src/pca_report_library/templates/generate_template.py`, `src/pca_report_library/customer/generate_report.py`, and `src/pca_report_library/compiler/xelatex.py`.
> 
> Please remove the executable bit from `src/pca_report_library/templates/generate_template.py` and `src/pca_report_library/customer/generate_report.py`.
> 
> All hard path references to assets need to be changed to use the [pkg_resources module](https://setuptools.readthedocs.io/en/latest/pkg_resources.html) like it is done in [cisagov/skeleton-python-library](https://github.com/cisagov/skeleton-python-library/blob/4ef2364701c0db074c423f740062b2d7c197bea8/src/example/example.py#L95-L99).
> 

resolved in b239b75beb335ee77d4a1e9168a4778ce35cd7da
> I would like any non-Python files inside the package file structure (`src/pca_report_library`) to be moved into `src/pca_report_library/assets/` to go hand-in-hand with the above change.

resolved in ab6bc6faa235ffc8f6a6bdcd2b0170ac72497b01